Kafka resize major upgrade + remove the workaround from values (#2411)
PostHog/posthog.com: contents/docs/self-host/deploy/upgrade-notes.md

    * to automatically install and manage them as part of your Helm release, simply leave the new `cert-manager.installCRDs` variable set to `true`
    * to manually install and manage them, simply set the new `cert-manager.installCRDs` variable to `false`

### Upgrading from 4.x.x

> 4.0.0 changes defaults for Kafka pvc sice and log retention policy. If your upgrade fails with the following:

Reviewer: ```suggestion
4.0.0 changes defaults for Kafka PVC size and log retention policy. If your upgrade fails with the following:
```

```
Error: UPGRADE FAILED: cannot patch "posthog-posthog-kafka" with kind StatefulSet: StatefulSet.apps "posthog-posthog-kafka" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```
There are two options:

> 1. Change your values to match to what the Kafka values were before (`size: 5Gi` and `logRetentionBytes: _4_000_000_000`)

Reviewer: ```suggestion
1. Change your values to match to what the Kafka values were before (`kafka.persistence.size: 20Gi` and `kafka.logRetentionBytes: _4_000_000_000`)
```

Reviewer: Wasn't 20Gi the default? Where did we remove `kafka.logRetentionBytes`?

Author: There are two groups of people here:
1. folks, who installed kafka before I added 20Gi default to values file, they should set `size: 5Gi` and `logRetentionBytes: _4_000_000_000` as the new `logRetentionBytes` default would be too high.
2. folks who used the values recently with 20Gi, they shouldn't see this error upon update, but their `retentionBytes` should get updated (I forgot it in the docs initially). I just tested on DO that's the case.

> 2. Resize the Kafka disk following our runbook [kafka-resize-disk](https://posthog.com/docs/self-host/runbook/kafka/resize-disk)

Reviewer: ```suggestion
2. Resize the PVC (Persistent Volume Claim) used by Kafka following our runbook [kafka-resize-disk](https://posthog.com/docs/self-host/runbook/kafka/resize-disk)
```
